chain/network/src/peer_manager/network_state/tier1.rs

```diff
@@ -27,16 +39,106 @@ impl super::NetworkState {
         self: &Arc<Self>,
         clock: &time::Clock,
     ) -> Vec<Arc<SignedAccountData>> {
+        // Tier1 advertise proxies calls should be disjoint,
+        // to avoid a race condition while connecting to the proxies.
+        // TODO(gprusak): there are more corner cases to cover, because
+        // tier1_connect may also spawn TIER1 connections conflicting with
+        // tier1_advertise_proxies. It would be better to be able to await
+        // handshake on connection attempts, even if another call spawned them.
+        let _lock = self.tier1_advertise_proxies_mutex.lock().await;
         let accounts_data = self.accounts_data.load();
-        let Some(vc) = self.tier1_validator_config(&accounts_data) else {
-            return vec![];
+        let tier1 = self.tier1.load();
+        let vc = match self.tier1_validator_config(&accounts_data) {
+            Some(it) => it,
+            None => {
+                tracing::info!(target:"network","This IS NOT a TIER1 node");
+                return vec![];
+            }
         };
-        // TODO(gprusak): for now we just blindly broadcast the static list of proxies, however
-        // here we should try to connect to the TIER1 proxies, before broadcasting them.
+        tracing::info!(target:"network","This IS a TIER1 node");
+        let proxies = match &vc.proxies {
+            config::ValidatorProxies::Dynamic(_) => {
+                // TODO(gprusak): If Dynamic are specified,
+                // it means that this node is its own proxy.
+                // Resolve the public IP of this node using those STUN servers,
+                // then connect to yourself (to verify the public IP).
+                vec![]
+            }
+            config::ValidatorProxies::Static(peer_addrs) => peer_addrs.clone(),
+        };
+        tracing::debug!(target:"test","proxies = {proxies:?}");
+        // Try to connect to all proxies in parallel.
+        let mut handles = vec![];
+        for proxy in proxies {
+            // Skip the proxies we are already connected to.
+            if tier1.ready.contains_key(&proxy.peer_id) {
+                continue;
+            }
+            handles.push(async move {
+                let res = async {
+                    let stream = tcp::Stream::connect(
+                        &PeerInfo {
+                            id: proxy.peer_id.clone(),
+                            addr: Some(proxy.addr),
+                            account_id: None,
+                        },
+                        tcp::Tier::T1,
+                    )
+                    .await?;
+                    anyhow::Ok(PeerActor::spawn_and_handshake(clock.clone(), stream, None, self.clone()).await?)
+                }.await;
+                if let Err(err) = res {
+                    tracing::info!(target:"network", ?err, "failed to establish connection to TIER1 proxy {:?}",proxy);
```

Comment:
shouldn't this be an error? (as the proxies are coming from the config right?)

Reply:
yes and no. The config might be outdated, the validator might be malicious. Changing to warn.